pipeline_dp/pipeline_operations.py

```diff
@@ -95,20 +90,50 @@ def is_serialization_immediate_on_reduce_by_key(self):
         return False
 
 
+class UniqueLabelsGenerator:
+    """Generate unique labels for each pipeline aggregation."""
+
+    def __init__(self, suffix):
+        self._labels = set()
+        self._suffix = suffix
+
+    def _add_if_unique(self, label):
+        if label in self._labels:
+            return False
+        self._labels.add(label)
+        return True
+
+    def unique(self, label):
+        if not label:
+            label = "UNDEFINED_STAGE_NAME"
+        suffix_label = label + "_" + self._suffix
+        if self._add_if_unique(suffix_label):
+            return suffix_label
+        for i in itertools.count(1):
+            label_candidate = label + "_" + str(i) + "_" + self._suffix
```

Comment:
@dvadym WDYT about appending the counter to after the label, instead of the suffix? It may be easier to trace a pipeline because the name would remain constant, eg.

Instead of `MAP_NAME_PIPELINE_SUFFIX_01` and `MAP_NAME_PIPELINE_SUFFIX _02`
We would have `MAP_NAME_1_PIPELINE_SUFFIX_0` and `MAP_NAME_2_PIPELINE_SUFFIX_0`

Does it make a difference when we want to build the execution graph? PTAL, thank you!